src/core/load-fragment.c

```diff
@@ -4730,26 +4730,47 @@ int config_parse_exec_directories(
                 /* For State and Runtime directories we support an optional destination parameter, which
                  * will be used to create a symlink to the source. */
                 _cleanup_free_ char *dresolved = NULL;
-                if (!isempty(dest)) {
+                ExecDirectoryFlags exec_directory_flags = 0;
+                if (!isempty(dest_or_flags)) {
+                        const char *destination = NULL;
+
                         if (streq(lvalue, "ConfigurationDirectory")) {
                                 log_syntax(unit, LOG_WARNING, filename, line, 0,
-                                           "Destination parameter is not supported for ConfigurationDirectory, ignoring: %s", tuple);
+                                           "Additional parameter is not supported for ConfigurationDirectory, ignoring: %s", tuple);
                                 continue;
                         }
 
-                        r = unit_path_printf(u, dest, &dresolved);
-                        if (r < 0) {
-                                log_syntax(unit, LOG_WARNING, filename, line, r,
-                                        "Failed to resolve unit specifiers in \"%s\", ignoring: %m", dest);
-                                continue;
+                        if (!isempty(flags)) {
+                                if (!streq(flags, "ro")) {
+                                        log_syntax(unit, LOG_WARNING, filename, line, 0,
+                                                   "Invalid flags for %s=, ignoring: %s", lvalue, flags);
+                                        continue;
+                                }
+
+                                destination = dest_or_flags;
+                                exec_directory_flags |= EXEC_DIRECTORY_READ_ONLY;
+                        } else {
+                                if (streq(dest_or_flags, "ro"))
```

Comment:
huh, what? what if i want to create a symlink called "ro"? that doesn't sound right?